examples/examples.test.ts

```diff
@@ -57,12 +57,17 @@ describe('Examples', () => {
       return;
     }
     const jsonSpec = JSON.parse(fs.readFileSync('examples/specs/' + example));
+    const stringifiedSpec = JSON.stringify(jsonSpec);
 
     describe(
       example,
       log.wrap(localLogger => {
         const vegaSpec: VgSpec = compile(jsonSpec).spec;
 
+        it('should not cause any side effects', () => {
+          expect(JSON.stringify(jsonSpec)).toEqual(stringifiedSpec);
```

Comment:
Shouldn't `const jsonSpec = duplicate(josnSpec); expect(jsonSpec).toEqual(originalJsonSpec)` work as well? It should show the diff in the tests so that would be good.